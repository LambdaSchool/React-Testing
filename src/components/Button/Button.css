.component-button {
    display: inline-flex;
    width: 25%;
    flex: 1 0 auto;
    height: 75px;
    margin: 1px 0;
}

.component-button.wide {
    width: 50%;
}

.component-button button {
    background-color: #E0E0E0;
    border: 0;
    font-size: 30px;
    margin: 0 1px 0 0;
    flex: 1 0 auto;
    padding: 0;
}

.component-button button:hover {
    background-color: rgb(184, 184, 184);
    cursor: pointer;
}

.component-button:last-child button {
    margin-right: 0;
}

.component-button.orange button {
    background-color: #F5923E;
    color: white;
}

.component-button.orange button:hover {
    background-color: rgb(209, 123, 49);
    cursor: pointer;
}