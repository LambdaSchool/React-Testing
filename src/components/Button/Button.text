Button 

props =>  orange  should be a boolean
... wide should be a boolean
... name shoul be a a string

function => handleClick accepts the name as input 




