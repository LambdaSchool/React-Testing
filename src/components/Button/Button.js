import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Button.css';

class Button extends Component {
    handleClick = () => {
        this.props.clickHandler(this.props.name);
    }

    render() {
        const classNames = [
            'component-button',
            this.props.orange ? 'orange' : '',
            this.props.wide ? 'wide' : '',
        ];
        return (
            <div className={classNames.join(' ').trim()}>
                <button onClick={this.handleClick}>
                    {this.props.name}
                </button>
            </div>
        );
    }
}
/* 
- renders the button with the appropriate classNames depedning on what props are passed
- should change the state for next if a number button has been pressed
*/
Button.propTypes = {
    name: PropTypes.string,
    orange: PropTypes.bool,
    wide: PropTypes.bool,
    clickHandler: PropTypes.func,
};

export default Button;