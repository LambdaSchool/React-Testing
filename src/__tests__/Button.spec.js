import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Button from '../components/Button/Button';

Enzyme.configure({ adapter: new Adapter() });

describe('<Button />', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button />, div);
  });

  it('renders a <div>', () => {
    const wrapper = shallow(<Button />);

    const elements = wrapper.find('div');

    expect(elements.length).toBe(1);

  });

  it('renders buttons correctly', () => {
    const tree = renderer
      .create(<Button name = {'ABC'} clickHandler =  { () => {} } />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick event on click of the button', () =>{
    
    const onClickTest = jest.fn();
    let wrapper = mount(<Button name={'buttonClickTest'} clickHandler={onClickTest}/>);
    wrapper.find('button').simulate('click');
    expect(onClickTest).toBeCalledWith('buttonClickTest');

  });


});
