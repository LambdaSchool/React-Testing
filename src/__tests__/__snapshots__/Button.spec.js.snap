// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders buttons correctly 1`] = `
<div
  className="component-button"
>
  <button
    onClick={[Function]}
  >
    ABC
  </button>
</div>
`;
